ext/include/opentelemetry/ext/http/client/curl/http_client_curl.h

```diff
@@ -280,18 +280,30 @@ class HttpClient : public opentelemetry::ext::http::client::HttpClient
 
   bool CancelAllSessions() noexcept override
   {
-    for (auto &session : sessions_)
+    // CancelSession may change sessions_, we can not change a container while iterating it.
+    while (!sessions_.empty())
     {
-      session.second->CancelSession();
+      std::map<uint64_t, std::shared_ptr<Session>> sessions;
+      sessions.swap(sessions_);
+      for (auto &session : sessions)
```

Comment:
why iterating over `sessions_` is not ok but iterating over `sessions` is ok here

Reply:
The call stack `HttpClient::CancelAllSessions` -> `Session::CancelSession` -> `HttpClient::CleanupSession` will crash here.
`CleanupSession` will change `sessions_` when iterating it.